src/realm/util/string_buffer.hpp

```diff
@@ -134,14 +132,16 @@ inline const char* StringBuffer::data() const noexcept
 
 inline char* StringBuffer::c_str() noexcept
 {
+    static char zero = 0;
```

Comment:
@danielpovlsen is right, this will worst case imply a mutex lock, best case an atomic instruction.

It would be better to let it be const, and fix `c_str()` to return a `const char*` (which is what `std::string` does).